streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -164,7 +165,27 @@ public void closeSuspended(final boolean clean,
     public List<ConsumerRecord<byte[], byte[]>> update(final TopicPartition partition,
                                                        final List<ConsumerRecord<byte[], byte[]>> records) {
         log.trace("Updating standby replicas of its state store for partition [{}]", partition);
-        return stateMgr.updateStandbyStates(partition, records);
+        final long limit = stateMgr.offsetLimit(partition);
+
+        final List<ConsumerRecord<byte[], byte[]>> toRestore = new ArrayList<>(records.size());
+        List<ConsumerRecord<byte[], byte[]>> remainingRecords = null;
```

Comment:
Why not init with `new ArrayList<>(records.size())` and avoid the check in the `for` loop? Could be `final` than, too.

If required, we can also `return remainingRecords.isEmpty() ? null : remainingRecords;` -- not sure atm who calls the method and what the impact of returning and empty list vs `null` is.

Reply:
Yeah, I had that as an intermediate state, but I figured if we're going to do the check at the end and return null, we may as well just keep initializing it to null. I think there's only one live caller, so it would be easy to migrate away from null semantics, though.

Reply:
Might be worth avoiding `null` -- no `null`, no NPE :)